packages/styles/src/components/header/index.scss

```diff
@@ -9,51 +9,56 @@
 
 tokens.$default-map: elements.$post-link;
 
-[data-post-scroll-locked] {
-  overflow: hidden !important;
-  transform: translate3d(0, 0, 0) !important;
-}
-
-post-header {
-  --header-expanded-height: calc(var(--global-header-height) + var(--local-header-height));
-  --header-reduced-height: calc(
-    var(--global-header-reduced-height) + var(--main-navigation-height)
+:root:has(post-header) {
+  --post-header-expanded-height: calc(
+    var(--post-global-header-height) + var(--post-local-header-height)
   );
-  --header-height: calc(
-    var(--header-expanded-height) -
-      min(
-        var(--header-scroll-top, 0px),
-        calc(var(--header-expanded-height) - var(--header-reduced-height))
+  --post-header-reduced-height: calc(
+    var(--post-global-header-reduced-height) + var(--post-main-navigation-height)
+  );
+  --post-header-height: calc(
+    var(--post-header-expanded-height) - min(
+        var(--post-header-scroll-top, 0px),
+        calc(var(--post-header-expanded-height) - var(--post-header-reduced-height))
       )
   );
-  height: var(--header-height);
+  scroll-padding-top: var(--post-header-height);
```

Comment:
We usually use the the root for custom properties only, I would expect the `scroll-padding-top` to be set on the body instead.